File: assignment7.R
# 4/18/23

# Importing the dataset
dataset = read.csv('Iris_Data.csv')

# Encoding categorical data
dataset$Species = factor(dataset$Species,
                         levels = c('Iris-setosa', 'Iris-versicolor', 'Iris-virginica'),
                         labels = c(1, 2, 3))

# Feature Scaling
scaled_cols = scale(dataset[, 1:2])
dataset[, 1:2] = scaled_cols

# Fitting SVM to the Training set
library(e1071)
classifier = svm(formula = Species ~ .,
                 data = dataset,
                 type = 'C-classification',
                 kernel = 'radial')

# Predicting the Test set results
y_pred = predict(classifier, newdata = dataset)

# Showing the Confusion Matrix and Accuracy
library(caret)
cm = confusionMatrix(y_pred, dataset$Species)
print(cm$table)
print(cm$overall['Accuracy'])

# Visualizing the  dataset results
set = dataset
X1 = seq(min(set[, 1]) - 1, max(set[, 1]) + 1, by = 0.01)
X2 = seq(min(set[, 2]) - 1, max(set[, 2]) + 1, by = 0.01)
grid_set = expand.grid(X1, X2)
colnames(grid_set) = c('Sepal.Length', 'Sepal.Width')
y_grid = predict(classifier, newdata = grid_set)
plot(NULL,
     main = 'SVM (Training set)',
     xlab = 'Sepal.Width (Scaled)', ylab = 'Sepal.Length (Scaled)',
     xlim = range(X1), ylim = range(X2))
points(grid_set, pch = 20, col = c('tomato', 'springgreen3', 'gold')[y_grid])
points(set, pch = 21, bg = c('red3', 'green4', 'yellow')[set$Species])

File: assignment7.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# =============================================================================
# Ayomide Ayowole-Obi
# Data Mining & Visualization 
# Assignment 7
# 4/18/23
# =============================================================================

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Iris_Data.csv')
X = dataset.iloc[:, :-1].to_numpy()
y = dataset.iloc[:, -1].to_numpy()

# Encoding the Dependent Variable - species
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
y = le.fit_transform(y)

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_scale = sc.fit_transform(X)

from sklearn.svm import SVC
classifier = SVC(kernel = 'rbf')
classifier.fit(X_scale, y)

# Predicting the Test set results
y_pred = classifier.predict(X_scale)

# Showing the Confusion Matrix and Accuracy
from sklearn.metrics import confusion_matrix, accuracy_score
print(confusion_matrix(y, y_pred))
print(accuracy_score(y, y_pred))

# =============================================================================
# RBF
# [[49  1  0]
#  [ 0 36 14]
#  [ 0 12 38]]
# 0.82
# =============================================================================

# Visualizing the dataset results
from matplotlib.colors import ListedColormap
X_set, y_set = X_scale, y
X1, X2 = np.meshgrid(np.arange(start = X_set[:, 0].min() - 1,
                               stop = X_set[:, 0].max() + 1, step = 0.01),
                     np.arange(start = X_set[:, 1].min() - 1,
                               stop = X_set[:, 1].max() + 1, step = 0.01))
plt.contourf(X1, X2, classifier.
             predict(np.array([X1.flatten(), X2.flatten()]).T).
             reshape(X1.shape),
             alpha = 0.75, cmap = ListedColormap(('red', 'green', 'gold')))
for i in np.unique(y_set):
    plt.scatter(X_set[y_set == i, 0], X_set[y_set == i, 1],
                color = ListedColormap(('brown', 'palegreen', 'yellow'))(i),
                edgecolors = 'black')
plt.title('SVM (Training set)')
plt.xlabel('Sepal Width (Scaled)')
plt.ylabel('Sepal Length (Scaled)')
plt.show()




